web/Controllers/PropertyController.cs

```diff
@@ -140,6 +140,32 @@ public ActionResult AddNewProperty_Cancel()
         // Go to View Properties page
         return RedirectToAction("ViewProperties", "Landlord");
     }
+    
+    [HttpPost]
+    [Authorize(Roles = "Landlord")]
+    [Authorize(Roles = "Admin")]
+    public ActionResult DeleteProperty(int propertyId)
+    {
+        PropertyDbModel? propDB = _propertyService.GetPropertyByPropertyId(propertyId);
+        if (propDB == null)
+        {
+            _logger.LogWarning("Property with ID {PropertyId} does not exist", propertyId);
+            return StatusCode(404);
+        }
+
+        if (GetCurrentUser().IsAdmin == false & GetCurrentUser().LandlordId != propDB.LandlordId)
+        {
+            _logger.LogWarning("Property with ID {PropertyId} is not your property.", propertyId);
+            return StatusCode(405);

```

Comment:
405 is method not allowed (e.g. you did a POST but this endpoint only accepts GET).

This should either be 403 Forbidden, or 404 Not Found if we don't want other users to even be able to determine whether the property exists or not.